src/dymaptic.GeoBlazor.Core/Scripts/openStreetMapLayer.ts

```diff
@@ -1,21 +1,73 @@
 // override generated code in this file
 import OpenStreetMapLayerGenerated from './openStreetMapLayer.gb';
 import OpenStreetMapLayer from '@arcgis/core/layers/OpenStreetMapLayer';
+import { copyValuesIfExists, hasValue } from "./arcGisJsInterop";
+
 
 export default class OpenStreetMapLayerWrapper extends OpenStreetMapLayerGenerated {
 
     constructor(layer: OpenStreetMapLayer) {
         super(layer);
     }
-    
+
 }
 
 export async function buildJsOpenStreetMapLayer(dotNetObject: any, layerId: string | null, viewId: string | null): Promise<any> {
-    let { buildJsOpenStreetMapLayerGenerated } = await import('./openStreetMapLayer.gb');
-    return await buildJsOpenStreetMapLayerGenerated(dotNetObject, layerId, viewId);
-}     
+    const { buildJsOpenStreetMapLayerGenerated } = await import('./openStreetMapLayer.gb');
+
+    const hasUrlTemplate = await prepareUrlTemplate(dotNetObject);
+
+    const osmLayer = await buildJsOpenStreetMapLayerGenerated(dotNetObject, layerId, viewId)
+
+    if (hasUrlTemplate) {
+        return createWebTileLayerFromOsmLayer(dotNetObject, layerId, viewId, osmLayer);
+    }
+
+    return osmLayer;
```

Comment:
I think the logic here all looks fine, but it needs some comments so the next dev looking at it knows why it isn't following the typical path, especially if ESRI fixes the underlying issue with ArcGIS OpenStreetMapLayer and we want to remove this later.